src/librustdoc/html/static/js/search.js

```diff
@@ -320,215 +733,220 @@ window.initSearch = function(rawSearchIndex) {
                     path = result.item.path.toLowerCase(),
                     parent = result.item.parent;
 
-                if (!isType && !validateResult(name, path, split, parent)) {
+                if (!isType && !validateResult(name, path, nameSplit, parent)) {
                     result.id = -1;
                 }
             }
             return transformResults(results);
         }
 
-        function extractGenerics(val) {
-            val = val.toLowerCase();
-            if (val.indexOf("<") !== -1) {
-                var values = val.substring(val.indexOf("<") + 1, val.lastIndexOf(">"));
-                return {
-                    name: val.substring(0, val.indexOf("<")),
-                    generics: values.split(/\s*,\s*/),
-                };
+        /**
+         * This function checks if the object (`row`) generics match the given type (`elem`)
```

Comment:
```suggestion
         * This function checks if this item's (`row`'s) function parameters match a type expression from the query (`elem`).
```

This is based on my understanding of what's going on; if I'm wrong please correct me.

Reply:
I'm not sure to understand your correction. Let me write an example and then maybe you can find a way to improve my wording so everyone can understand. So let's take this function:

```rust
fn foo<T: Clone>(v: Vec<T>);
```

So in here, `checkGenerics` will be looking into the `T` from `Vec<T>` which has been transformed into `Clone`. "generics" here is a type's trait bounds.